prog.go

```diff
@@ -140,8 +140,10 @@ func newProgramWithBTF(spec *ProgramSpec, btf *btf.Handle, opts ProgramOptions)
 		_, logErr := bpfProgLoad(attr)
 		err = internal.ErrorWithLog(err, logBuf, logErr)
 	}
-
-	return nil, errors.Wrap(err, "can't load program")
+	if err != nil {
+		return nil, xerrors.Errorf("can't load program: %w", err)
```

Comment:
We already know that err != nil, so this if isn't needed. Just `return nil, xerrors.Errorf("can't load program: %w", err)`

Reply:
`errors.Wrap()` returns [1] nil, if `err == nil`. Otherwise it returns the annotated error. So, this can not be simply replaced by `xerrors.Errorf()`.

[1] https://godoc.org/github.com/pkg/errors#Wrap

Reply:
There is a check for `if err == nil` further up, so if we end up at this point `err != nil` always.

Reply:
correct :see_no_evil: this not see this before. will adapt it.